app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -2180,91 +2122,51 @@ class BrowserTabFragment :
     }
 
     private fun configureWebViewForAutofill(it: DuckDuckGoWebView) {
-        browserAutofill.addJsInterface(it, autofillCallback, this, null, tabId)
+        launch(dispatchers.main()) {
+            browserAutofill.addJsInterface(it, autofillCallback, tabId)
 
-        autofillFragmentResultListeners.getPlugins().forEach { plugin ->
-            setFragmentResultListener(plugin.resultKey(tabId)) { _, result ->
-                context?.let {
-                    plugin.processResult(
-                        result = result,
-                        context = it,
-                        tabId = tabId,
-                        fragment = this@BrowserTabFragment,
-                        autofillCallback = this@BrowserTabFragment,
-                    )
+            autofillFragmentResultListeners.getPlugins().forEach { plugin ->
+                setFragmentResultListener(plugin.resultKey(tabId)) { _, result ->
+                    context?.let {
+                        plugin.processResult(
+                            result = result,
+                            context = it,
+                            tabId = tabId,
+                            fragment = this@BrowserTabFragment,
+                            autofillCallback = this@BrowserTabFragment,
+                        )
+                    }
                 }
             }
         }
     }
 
-    private fun injectAutofillCredentials(
-        url: String,
-        credentials: LoginCredentials?,
-    ) {
-        webView?.let {
-            if (it.url != url) {
-                Timber.w("WebView url has changed since autofill request; bailing")
-                return
-            }
-            browserAutofill.injectCredentials(credentials)
-        }
-    }
-
-    private fun acceptGeneratedPassword(url: String) {
-        webView?.let {
-            if (it.url != url) {
-                Timber.w("WebView url has changed since autofill request; bailing")
-                return
-            }
-            browserAutofill.acceptGeneratedPassword()
-        }
-    }
-
-    private fun rejectGeneratedPassword(url: String) {
-        webView?.let {
-            if (it.url != url) {
-                Timber.w("WebView url has changed since autofill request; bailing")
-                return
-            }
-            browserAutofill.rejectGeneratedPassword()
-        }
-    }
-
     private fun cancelPendingAutofillRequestsToChooseCredentials() {
         browserAutofill.cancelPendingAutofillRequestToChooseCredentials()
         viewModel.cancelPendingAutofillRequestToChooseCredentials()
     }
 
     private fun showAutofillDialogSaveCredentials(
-        currentUrl: String,
+        autofillUrlRequest: AutofillUrlRequest,
         credentials: LoginCredentials,
     ) {
-        val url = webView?.url ?: return
-        if (url != currentUrl) return
```

Comment:
note, deliberately removing the URL check here.

the offer to save a credential after logging in is ok to show over another page (it's typical to navigate immediately after submitting a form)